File: ResourceManager/Models/Configurations/NoteConfiguration.cs
using Microsoft.EntityFrameworkCore;
using Microsoft.EntityFrameworkCore.Metadata.Builders;

namespace ResourceManager.Models.Configurations
{
    public class NoteConfiguration : IEntityTypeConfiguration<Note>
    {
        public void Configure(EntityTypeBuilder<Note> builder)
        {
            builder.ToTable("Notes");
            builder.HasKey(x => x.ID);

            builder.Property(x => x.ID).ValueGeneratedOnAdd();
            builder.Property(x => x.CreatedDate).HasDefaultValueSql("date('now','localtime')");
            builder.Property(x => x.ModifiedDate).HasDefaultValueSql("date('now','localtime')");
            builder.Property(x => x.Title).HasColumnType("nvarchar(255)").IsRequired();
            builder.Property(x => x.Date).HasColumnType("text").IsRequired();
            builder.Property(x => x.Notes).HasColumnType("nvarchar(500)").IsRequired();
            builder.HasIndex(x => new { x.Date, x.Title }).IsUnique();

            builder.HasOne(x => x.User)
                .WithMany(x => x.Notes)
                .HasForeignKey(x => x.UserID);
        }
    }
}

File: ResourceManager/Models/Configurations/NoteConfiguration.vb
Imports Microsoft.EntityFrameworkCore
Imports Microsoft.EntityFrameworkCore.Metadata.Builders

Namespace ResourceManager.Models.Configurations
    Public Class NoteConfiguration
        Implements IEntityTypeConfiguration(Of Note)
        Public Sub Configure(ByVal builder As EntityTypeBuilder(Of Note)) Implements IEntityTypeConfiguration(Of Note).Configure
            builder.ToTable("Notes")
            builder.HasKey(Function(x) x.ID)

            builder.Property(Function(x) x.ID).ValueGeneratedOnAdd()
            builder.[Property](Function(x) x.CreatedDate).HasDefaultValueSql("date('now','localtime')")
            builder.[Property](Function(x) x.ModifiedDate).HasDefaultValueSql("date('now','localtime')")
            RelationalPropertyBuilderExtensions.HasColumnType(builder.Property(Function(x) x.Title), "nvarchar(255)").IsRequired()
            RelationalPropertyBuilderExtensions.HasColumnType(builder.Property(Function(x) x.Date), "text").IsRequired()
            RelationalPropertyBuilderExtensions.HasColumnType(builder.Property(Function(x) x.Notes), "nvarchar(500)").IsRequired()
            builder.HasIndex(CType(Function(x) New With {
                x.Date,
                x.Title
            }, Expressions.Expression(Of Func(Of Note, Object)))).IsUnique()

            builder.HasOne(Function(x) x.User).WithMany(Function(x) x.Notes).HasForeignKey(Function(x) x.UserID)
        End Sub
    End Class
End Namespace
